### PR TITLE
add info on hint() method

### PR DESCRIPTION
**Pull request by Xosofox**
hint was not mentioned before

### CODE REVIEW

File: docs/en/reference/query-builder-api.rst [hunk 1]
```diff
@@ -185,6 +185,25 @@ the ``select()`` method:
 In the results only the data from the username and password will be
 returned.
 
+Index hints
+~~~~~~~~~~~
+
+You can help mongo in finding the right index to use with the ``hint()`` method:
```
Comment (malarzm) [inline docs/en/reference/query-builder-api.rst]: `$hint` is rather forcing Mongo to use specified index, think it's worth reflecting and linking to https://docs.mongodb.com/manual/reference/operator/meta/hint/

File: docs/en/reference/query-builder-api.rst [hunk 2]
```diff
@@ -185,6 +185,25 @@ the ``select()`` method:
 In the results only the data from the username and password will be
 returned.
 
+Index hints
+~~~~~~~~~~~
+
+You can force mongo to use a specific index for a query with the ``hint()`` method (see `hint <https://docs.mongodb.com/manual/reference/operator/meta/hint/>`_)
```
Comment (alcaeus) [inline docs/en/reference/query-builder-api.rst]: You should use "MongoDB", not "mongo"

Comment (Xosofox) [inline docs/en/reference/query-builder-api.rst]: You're right, that's not very polite

**Comment by malarzm** [comment]
@Xosofox LGTM, would be great if you could squash your commits (or I can do that later today)

**Comment by malarzm** [comment]
Thanks @Xosofox!

**Pull request merged**